src/runtime/typemanager.cs

```diff
@@ -784,21 +816,25 @@ public SlotsHolder(IntPtr type)
 
         public void Set(int offset, ThunkInfo thunk)
         {
+            if (_slots == null) _slots = new Dictionary<int, ThunkInfo>();
```

Comment:
This is not thread safe. I am unsure holding GIL ensures this is properly synchronized.

Reply:
You can assume it's safe when you holding the GIL, all these slots stuff are linear behaviors for the type initialization.